### PR TITLE
Decode secret internal data into struct and fix type assertion.

### LINKED ISSUE #1779: secret/ssh: panic on unmount
When trying to unmount ssh backend, I get `Unmount error: EOF`.
Looking in the logs yields a server panic.

## Info
- Client: 0.6.1
- Server: 0.6.1

## Crash Log
```
vault[1111]: http: panic serving 172.31.255.4:52055: interface conversion: interface is json.Number, not float64
goroutine 2042370 [running]:
net/http.(*conn).serve.func1(0xc421414380)
	   /goroot/src/net/http/server.go:1491 +0x12a
panic(0x129d340, 0xc421222600)
	   /goroot/src/runtime/panic.go:458 +0x243
github.com/hashicorp/vault/builtin/logical/ssh.(*backend).secretDynamicKeyRevoke(0xc4203a0d50, 0xc420f60280, 0xc420f4f210, 0x12f2a80, 0x68f901, 0xc420f4f210)
	   /gopath/src/github.com/hashicorp/vault/builtin/logical/ssh/secret_dynamic_key.go:95 +0x50b
github.com/hashicorp/vault/builtin/logical/ssh.(*backend).(github.com/hashicorp/vault/builtin/logical/ssh.secretDynamicKeyRevoke)-fm(0xc420f60280, 0xc420f4f210, 0x0, 0x2, 0xc420512690)
	   /gopath/src/github.com/hashicorp/vault/builtin/logical/ssh/secret_dynamic_key.go:27 +0x3e
github.com/hashicorp/vault/logical/framework.(*Secret).HandleRevoke(0xc420512690, 0xc420f60280, 0x1460c4e, 0x6, 0xc420ef1701)
	   /gopath/src/github.com/hashicorp/vault/logical/framework/secret.go:86 +0x8b
github.com/hashicorp/vault/logical/framework.(*Backend).handleRevokeRenew(0xc4203c0300, 0xc420f60280, 0x1460c4e, 0x6, 0x1)
	   /gopath/src/github.com/hashicorp/vault/logical/framework/backend.go:407 +0x23d
github.com/hashicorp/vault/logical/framework.(*Backend).HandleRequest(0xc4203c0300, 0xc420f60280, 0xc420f60280, 0xc420f4ee90, 0xe)
	   /gopath/src/github.com/hashicorp/vault/logical/framework/backend.go:152 +0x7dd
github.com/hashicorp/vault/vault.(*Router).routeCommon(0xc4201a8540, 0xc420f60280, 0x40ef00, 0x0, 0x1410000, 0x0, 0x0)
	   /gopath/src/github.com/hashicorp/vault/vault/router.go:269 +0x748
github.com/hashicorp/vault/vault.(*Router).Route(0xc4201a8540, 0xc420f60280, 0x10c0, 0x1280e40, 0xc4218be750)
	   /gopath/src/github.com/hashicorp/vault/vault/router.go:187 +0x3a
github.com/hashicorp/vault/vault.(*ExpirationManager).revokeEntry(0xc4204feec0, 0xc4218be750, 0x33, 0xc4218be750)
	   /gopath/src/github.com/hashicorp/vault/vault/expiration.go:598 +0x22c
github.com/hashicorp/vault/vault.(*ExpirationManager).revokeCommon(0xc4204feec0, 0xc421222540, 0x33, 0xc420c20000, 0x0, 0x0)
	   /gopath/src/github.com/hashicorp/vault/vault/expiration.go:201 +0x351
github.com/hashicorp/vault/vault.(*ExpirationManager).revokePrefixCommon(0xc4204feec0, 0xc420c2b366, 0x4, 0x0, 0x2, 0xecf4f77bb)
	   /gopath/src/github.com/hashicorp/vault/vault/expiration.go:297 +0x1f5
github.com/hashicorp/vault/vault.(*ExpirationManager).RevokePrefix(0xc4204feec0, 0xc420c2b366, 0x4, 0x0, 0x0)
	   /gopath/src/github.com/hashicorp/vault/vault/expiration.go:246 +0x10d
github.com/hashicorp/vault/vault.(*Core).unmount(0xc4200ee380, 0xc420c2b366, 0x4, 0x0, 0x0)
	   /gopath/src/github.com/hashicorp/vault/vault/mount.go:265 +0x3a5
github.com/hashicorp/vault/vault.(*SystemBackend).handleUnmount(0xc420474d00, 0xc420f600a0, 0xc420f4ec20, 0x2000107, 0x0, 0xffffffffffffffff)
	   /gopath/src/github.com/hashicorp/vault/vault/logical_system.go:809 +0x13d
github.com/hashicorp/vault/vault.(*SystemBackend).(github.com/hashicorp/vault/vault.handleUnmount)-fm(0xc420f600a0, 0xc420f4ec20, 0x0, 0x6, 0xc420527000)
	   /gopath/src/github.com/hashicorp/vault/vault/logical_system.go:227 +0x3e
github.com/hashicorp/vault/logical/framework.(*Backend).HandleRequest(0xc4203ba240, 0xc420f600a0, 0xc420f600a0, 0xc420c2b35b, 0xf)
	   /gopath/src/github.com/hashicorp/vault/logical/framework/backend.go:206 +0x42e
github.com/hashicorp/vault/vault.(*Router).routeCommon(0xc4201a8540, 0xc420f600a0, 0x0, 0x0, 0x0, 0x0, 0x0)
	   /gopath/src/github.com/hashicorp/vault/vault/router.go:269 +0x748
github.com/hashicorp/vault/vault.(*Router).Route(0xc4201a8540, 0xc420f600a0, 0xc420f600a0, 0x0, 0x0)
	   /gopath/src/github.com/hashicorp/vault/vault/router.go:187 +0x3a
github.com/hashicorp/vault/vault.(*Core).handleRequest(0xc4200ee380, 0xc420f600a0, 0x0, 0x0, 0x0, 0x0)
	   /gopath/src/github.com/hashicorp/vault/vault/request_handling.go:160 +0xb90
github.com/hashicorp/vault/vault.(*Core).HandleRequest(0xc4200ee380, 0xc420f600a0, 0x0, 0x0, 0x0)
	   /gopath/src/github.com/hashicorp/vault/vault/request_handling.go:42 +0x687
github.com/hashicorp/vault/http.request(0xc4200ee380, 0x1ced5c0, 0xc420e5bad0, 0xc4214264b0, 0xc420f600a0, 0x0, 0x0)
	   /gopath/src/github.com/hashicorp/vault/http/handler.go:161 +0x39
github.com/hashicorp/vault/http.handleLogical.func1(0x1ced5c0, 0xc420e5bad0, 0xc4214264b0)
	   /gopath/src/github.com/hashicorp/vault/http/logical.go:110 +0x10e
net/http.HandlerFunc.ServeHTTP(0xc420492780, 0x1ced5c0, 0xc420e5bad0, 0xc4214264b0)
	   /goroot/src/net/http/server.go:1726 +0x44
github.com/hashicorp/vault/http.handleRequestForwarding.func1(0x1ced5c0, 0xc420e5bad0, 0xc4214264b0)
	   /gopath/src/github.com/hashicorp/vault/http/handler.go:128 +0x696
net/http.HandlerFunc.ServeHTTP(0xc4204927c0, 0x1ced5c0, 0xc420e5bad0, 0xc4214264b0)
	   /goroot/src/net/http/server.go:1726 +0x44
net/http.(*ServeMux).ServeHTTP(0xc4201a8960, 0x1ced5c0, 0xc420e5bad0, 0xc4214264b0)
	   /goroot/src/net/http/server.go:2022 +0x7f
github.com/hashicorp/vault/http.handleHelpHandler.func1(0x1ced5c0, 0xc420e5bad0, 0xc4214264b0)
	   /gopath/src/github.com/hashicorp/vault/http/help.go:18 +0xd8
net/http.HandlerFunc.ServeHTTP(0xc420492860, 0x1ced5c0, 0xc420e5bad0, 0xc4214264b0)
	   /goroot/src/net/http/server.go:1726 +0x44
net/http.serverHandler.ServeHTTP(0xc42049a300, 0x1ced5c0, 0xc420e5bad0, 0xc4214264b0)
	   /goroot/src/net/http/server.go:2202 +0x7d
net/http.(*conn).serve(0xc421414380, 0x1cee6c0, 0xc420e8d480)
	   /goroot/src/net/http/server.go:1579 +0x4b7
created by net/http.(*Server).Serve
	   /goroot/src/net/http/server.go:2293 +0x44d
```

### PR DESCRIPTION
**Pull request by jefferai**
Fixes #1779

### CODE REVIEW

File: builtin/logical/ssh/secret_dynamic_key.go [hunk 1]
```diff
@@ -34,78 +36,34 @@ func (b *backend) secretDynamicKeyRenew(req *logical.Request, d *framework.Field
...
-		return nil, fmt.Errorf("secret is missing internal data")
-	}
-
-	portRaw, ok := req.Secret.InternalData["port"]
-	if !ok {
-		return nil, fmt.Errorf("secret is missing internal data")
+	intSec := &sec{}
+	err := mapstructure.Decode(req.Secret.InternalData, intSec)
+	if err != nil {
+		return nil, errwrap.Wrapf("secret internal data could not be decoded: {{err}}", err)
```
Comment (vishalnayak) [inline builtin/logical/ssh/secret_dynamic_key.go]: Any reason for using `errwrap` here instead of `fmt.Errorf`?

**Comment by vishalnayak** [comment]
One minor optional comment, other than that LGTM!

**Comment by BSick7** [comment]
Whoa. This is quite different. I love it.

File: builtin/logical/ssh/secret_dynamic_key.go (hunk 1 above)
Comment (jefferai) [inline builtin/logical/ssh/secret_dynamic_key.go]: (slack)

**Comment by LLBennett** [comment]
Looks great!

**Pull request merged**